[BA-2207] Adds script to generate jwks well known file & the private key

sakethkotamraju opened the pull request:
### _Summary_

https://linear.app/coinbase/issue/BA-2207/wsdk-create-script-to-generate-well-known-jwks-file-as-well-as-the

Generates the  sec256k1 public key & outputs it in the base-jwks.json file. It also outputs the corresponding private key in the domain-verification-private-key.txt file. I also made a docs/domain-verification-key-generation readme file for instructions on how to do all of this.

### _How did you test your changes?_

I ran the command and verified the generated output contents were accurate and outputted in the correct folder/file path. 

https://github.com/user-attachments/assets/4730cc53-9917-4581-9709-28d38a6bff5d

sakethkotamraju committed:
key generation script for well known file

- domain verification key generation script for well known file
- yarn generate-key-script
- readme doc on how to use it

sakethkotamraju committed:
Update README.md

sakethkotamraju committed:
changed from jwks.json to base-jwks.json

montycheese commented on packages/client/scripts/generate-domain-keys.ts:
```diff
@@ -0,0 +1,96 @@
+
+function getOutputDir(): string {
+  // Check for --cwd argument
+  const cwdArgIndex = process.argv.indexOf('--cwd');
+  if (cwdArgIndex !== -1 && process.argv[cwdArgIndex + 1]) {
+    return process.argv[cwdArgIndex + 1];
```
this isn't documented anywhere - but we likely don't  need it anyways.

montycheese commented on packages/client/scripts/generate-domain-keys.ts:
```diff
@@ -0,0 +1,96 @@
+    x,
+    y,
+    use: 'sig',
+    kid: 'coinbase-domain-verification',
```
base

montycheese commented on packages/client/scripts/generate-domain-keys.ts:
```diff
@@ -0,0 +1,96 @@
+  try {
+    const outputDir = getOutputDir();
+
+    console.log('🔑 Generating Coinbase domain verification keys...\n');
```
Base

montycheese commented on packages/client/scripts/generate-domain-keys.ts:
```diff
@@ -0,0 +1,96 @@
+    const { jwks, privateKey } = generateDomainVerificationKeys();
+    
+    // Create .well-known directory if it doesn't exist
+    const wellKnownDir = join(outputDir, '.well-known');
+    mkdirSync(wellKnownDir, { recursive: true });
```
I don't think we need to output it to a local directory called .well-known. also its currently but created as a hidden directory which might be annoying

montycheese reviewed (commented):
left some comments

sakethkotamraju committed:
fixed changes for pr

- no .well-known output directory
- changed 'Coinbase' to 'base'

montycheese commented on packages/client/scripts/generate-domain-keys.ts:
```diff
@@ -0,0 +1,87 @@
+
+function main() {
+  try {
+    console.log('🔑 Generating Coinbase domain verification keys...\n');
```
```suggestion
    console.log('🔑 Generating Base domain verification keys...\n');
```

montycheese reviewed (commented):
Lgtm. One small update. I'm going to pull locally to verify running locally before approving

sakethkotamraju committed:
signature verification script

montycheese commented on packages/client/scripts/generate-domain-keys.ts:
```diff
@@ -0,0 +1,87 @@
+function generateDomainVerificationKeys() {
+  // Generate a random private key
+  const privateKeyBytes = randomBytes(32);
+  const privateKey = base64url(privateKeyBytes);
```
Does this need to be in base 64?

montycheese commented on verify-signature.js:
```diff
@@ -0,0 +1,68 @@
+const crypto = require('crypto');
```
move this to scripts/verify-domain-keys-test.js

fan-zhang-sv commented on docs/domain-verification-key-generation.md:
````diff
@@ -0,0 +1,101 @@
+```
+
+## Output
+- `.well-known/base-jwks.json`: Public key in JWKS format for domain verification
````
i think the output here is just `base-jwk.json`? cuz the file is generated in `projectRoot/base-jek.json`

sakethkotamraju committed:
validation script fixes

- doesn't base64 encode the private key
- yarn validate-key-script for testing sec256kp1 encoding/decoding.

sakethkotamraju committed:
fixed output file name

sakethkotamraju committed:
updated yarn.lock to match package.json

sakethkotamraju committed:
removed print statements for yarn lint
